### PR TITLE
en,zh: in k8s -> on k8s

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added, or deleted? (Required)
ref: https://github.com/pingcap/docs/pull/11224

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v1.4 (TiDB Operator 1.4 versions)
- [x] v1.3 (TiDB Operator 1.3 versions)
- [ ] v1.2 (TiDB Operator 1.2 versions)
- [ ] v1.1 (TiDB Operator 1.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s):

### CODE REVIEW

File: en/deploy-on-general-kubernetes.md [hunk 1]
```diff
@@ -124,7 +124,7 @@ Different clusters can be in the same or different `namespace`, which is based o
 
 ## Initialize the TiDB cluster
 
-If you want to initialize your cluster after deployment, refer to [Initialize a TiDB Cluster in Kubernetes](initialize-a-cluster.md).
```
Comment (Oreoxmt) [inline en/deploy-on-general-kubernetes.md:127]: `Deploy TiDB in General Kubernetes` -> `Deploy TiDB on General Kubernetes`

https://github.com/ran-huang/docs-tidb-operator/blob/on-k8s/en/deploy-multiple-tidb-operator.md?plain=1#L58

```
2. Refer to [Deploy TiDB on General Kubernetes](deploy-on-general-kubernetes.md) to deploy the TiDB cluster.
```

File: en/TOC.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <!-- markdownlint-disable MD007 -->
 <!-- markdownlint-disable MD041 -->
 
-- [TiDB in Kubernetes Docs](https://docs.pingcap.com/tidb-in-kubernetes/dev)
+- [TiDB on Kubernetes Docs](https://docs.pingcap.com/tidb-in-kubernetes/dev)
```
Comment (Oreoxmt) [inline en/TOC.md:4]: I'm not sure whether L10 and L18 need to be changed to `On ... Kubernetes`

**Review by shichun-0415 — APPROVED** [review]

File: en/TOC.md (hunk 2 above)
Comment (ran-huang) [reply en/TOC.md:4]: Nice catch. Updated.

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Pull request merged**